### PR TITLE
IGNORE - FOR TESTING PURPOSES

### PR DESCRIPTION
**Pull request by PavlosTze**
IGNORE - FOR TESTING PURPOSES

### CODE REVIEW

**Comment by PavlosTze** [comment]
## Code Review Agent Run Status

- **Limitations and other issues:** <small>❌</small>&nbsp;Failure - The AI Code Review Agent skipped reviewing this change because it is configured to exclude certain pull requests based on the source/target branch or the pull request status. You can change the settings [here](https://alpha.bito.ai/home/ai-agents/configured-agents/1), or contact the agent instance creator at [pavlos@weatherxm.com](mailto:pavlos@weatherxm.com).

**Comment by PavlosTze** [comment]
AI Code Review is in progress (usually takes <3 minutes unless it's a very large PR).

File: app/src/main/java/com/weatherxm/util/NumberUtils.kt [hunk 1]
```diff
@@ -1,35 +1,61 @@
...
 import java.math.RoundingMode
+import java.text.DecimalFormat
+import java.text.NumberFormat
 
 object NumberUtils : KoinComponent {
     private val compactDecimalFormat: CompactDecimalFormat by inject()
     private val numberFormat: NumberFormat by inject()
+    private val decimalFormat: DecimalFormat by inject(named(DECIMAL_FORMAT_TOKENS))
+
+    private const val DIVISOR_WEI_TO_ETH = "1000000000000000000"
```
Comment (PavlosTze) [inline app/src/main/java/com/weatherxm/util/NumberUtils.kt:19]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/350#issuecomment-2388457789>#b046d2</a><span> - 10/02/2024, 12:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The DIVISOR_WEI_TO_ETH constant is defined as a String, which may lead to inefficient operations and potential precision loss when used in calculations involving Wei to ETH conversions.
</div>
<div id="fix">
<b>Fix:</b> Replace the String constant with a BigInteger or BigDecimal constant to ensure precise calculations when dealing with Wei to ETH conversions.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,2 +19,2 @@ object NumberUtils : KoinComponent {
 -    private const val DIVISOR_WEI_TO_ETH = "1000000000000000000"
 +    private val DIVISOR_WEI_TO_ETH = BigInteger("1000000000000000000")
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: app/src/test/java/com/weatherxm/util/ChartsAxisFormattersTest.kt [hunk 2]
```diff
@@ -2,13 +2,30 @@ package com.weatherxm.util
...
+            modules(
+                module {
+                    single<NumberFormat> {
+                        NumberFormat.getInstance(Locale.US)
+                    }
+                }
+            )
+        }
+    }
+
```
Comment (PavlosTze) [inline app/src/test/java/com/weatherxm/util/ChartsAxisFormattersTest.kt:28]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/350#issuecomment-2388457789>#b046d2</a><span> - 10/02/2024, 12:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The Koin setup in the beforeSpec block initializes for all tests, but there's no corresponding afterSpec block to stop Koin. Similar issues were also found in:
- app/src/test/java/com/weatherxm/util/ChartsAxisFormattersTest.kt (line 27)
- app/src/test/java/com/weatherxm/util/NumberUtilsTest.kt (line 23)
</div>
<div id="fix">
<b>Fix:</b> Add an afterSpec block to stop Koin after all tests have completed.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -27,6 +27,10 @@ class ChartsAxisFormattersTest : BehaviorSpec({
      }
  }

 +    afterSpec {
 +        stopKoin()
 +    }
 +
  context("Use custom X axis formatter") {
      given("A list of times") {
          then("it should return the correct time") {
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: app/src/test/java/com/weatherxm/util/ChartsAxisFormattersTest.kt [hunk 3]
```diff
@@ -40,4 +57,8 @@ class ChartsAxisFormattersTest : BehaviorSpec({
             }
         }
     }
+
+    afterSpec {
```
Comment (PavlosTze) [inline app/src/test/java/com/weatherxm/util/ChartsAxisFormattersTest.kt:61]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/350#issuecomment-2388457789>#b046d2</a><span> - 10/02/2024, 12:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The afterSpec block is added correctly, but there's no corresponding startKoin() call visible in the provided context, which may lead to inconsistent test setup and teardown.
</div>
<div id="fix">
<b>Fix:</b> Add a corresponding startKoin() call in a beforeSpec block to properly initialize Koin before the tests run.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -3,6 +3,7 @@ import io.kotest.core.spec.style.BehaviorSpec
  import io.kotest.matchers.shouldBe
  import org.koin.core.context.startKoin
  import org.koin.core.context.stopKoin
  import org.koin.dsl.module
  import java.text.NumberFormat
  import java.util.Locale
 
  class ChartsAxisFormattersTest : BehaviorSpec({
 +    beforeSpec {
 +        startKoin {
 +            modules(module {
 +                // Add your test modules here
 +            })
 +        }
 +    }
 
      // ... existing test code ...
 
      afterSpec {
          stopKoin()
      }
  })
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: app/src/main/java/com/weatherxm/ui/components/DailyRewardsCardView.kt [hunk 4]
```diff
@@ -76,10 +74,8 @@ open class DailyRewardsCardView : LinearLayout, KoinComponent {
         binding.baseRewardIcon.setColorFilter(
             context.getColor(getRewardScoreColor(data.baseRewardScore))
         )
-
-        binding.baseRewardScore.text = data.baseReward?.let {
-            context.getString(R.string.wxm_amount, formatTokens(it.toBigDecimal()))
-        } ?: EMPTY_VALUE
+        binding.baseRewardScore.text =
```
Comment (PavlosTze) [inline app/src/main/java/com/weatherxm/ui/components/DailyRewardsCardView.kt:77]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/350#issuecomment-2388457789>#b046d2</a><span> - 10/02/2024, 12:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Potential null pointer exception if data.baseReward is null.
</div>
<div id="fix">
<b>Fix:</b> Add a null check for data.baseReward before passing it to formatTokens function.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -77,2 +77,4 @@
 -        binding.baseRewardScore.text =
 -            context.getString(R.string.wxm_amount, formatTokens(data.baseReward))
 +        binding.baseRewardScore.text = context.getString(
 +            R.string.wxm_amount,
 +            if (data.baseReward != null) formatTokens(data.baseReward)
 +            else EMPTY_VALUE)
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: app/src/test/java/com/weatherxm/util/NumberUtilsTest.kt [hunk 5]
```diff
@@ -43,19 +58,74 @@ class NumberUtilsTest : BehaviorSpec({
...
+    context("Convert WEI -> ETH correctly and show the correct amount") {
+        given("a WEI amount") {
+            When("it is zero") {
+                then("the convertor should return zero") {
+                    weiToETH(BigDecimal.ZERO) shouldBe BigDecimal.ZERO
+                }
+            }
+            When("it is bigger than zero") {
+                then("the validator should return make the conversion and format the tokens") {
+                    formatTokens(weiToETH(BigDecimal.valueOf(10000000000000000))) shouldBe "0.01"
```
Comment (PavlosTze) [inline app/src/test/java/com/weatherxm/util/NumberUtilsTest.kt:128]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/350#issuecomment-2388457789>#b046d2</a><span> - 10/02/2024, 12:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The test case for WEI to ETH conversion only tests a single non-zero scenario, which may not be sufficient to ensure the function's correctness across different input ranges.
</div>
<div id="fix">
<b>Fix:</b> Add more test cases to cover a wider range of input values for the WEI to ETH conversion, including edge cases and boundary values.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -126,6 +126,18 @@ class NumberUtilsTest : BehaviorSpec({
          When("it is bigger than zero") {
              then("the validator should return make the conversion and format the tokens") {
                  formatTokens(weiToETH(BigDecimal.valueOf(10000000000000000))) shouldBe "0.01"
 +                // Test with 1 WEI (smallest unit)
 +                formatTokens(weiToETH(BigDecimal.ONE)) shouldBe "0.00"
 +                // Test with 1 ETH in WEI
 +                formatTokens(weiToETH(BigDecimal.valueOf(1000000000000000000))) shouldBe "1.00"
 +                // Test with a large WEI amount
 +                formatTokens(weiToETH(BigDecimal.valueOf(1234567890000000000000))) shouldBe "1,234.57"
 +                // Test with a very small non-zero WEI amount
 +                formatTokens(weiToETH(BigDecimal.valueOf(1000))) shouldBe "0.00"
 +            }
 +            and("the input is the maximum possible WEI value") {
 +                then("the conversion should not overflow") {
 +                    weiToETH(BigDecimal("115792089237316195423570985008687907853269984665640564039457584007913129639935")) shouldBe BigDecimal("115792089237316195423570985008687907853269984665640564039457.584007913129639935")
              }
          }
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged
